### DISCUSSION TITLE
How to make the root url redirect to two dynamically nested routes?

### DISCUSSION THREAD

**Discussion by makezi** [vercel/next.js#59617] [App Router]
Hello, was wondering if anyone could help with this issue I'm having learning the app router in regards to deeply nested dynamic routes as the root url.

Lets say I have an ecommerce shop which hosts multiple stores with their own available shopping modes (pickup and delivery) and would like to have the store id and shopping mode in the url: www.shop.com/store/STOREID/sm/SHOPPINGMODE. 

When first time users visit www.shop.com they are first asked which shop they want and shopping mode (saved locally as a cookie for example) and any future visits it will redirect them from the root to the `/store/STOREID/sm/SHOPPINGMODE` url as mentioned.

The other thing is that if the user has a store selected and they decide to go www.shop.com/products it will redirect them to www.shop.com/store/STOREID/sm/SHOPPINGMODE/products instead.

So basically:

`/`
* If first time visit: www.shop.com -> choose store -> www.shop.com/store/STOREID/sm/SHOPPINGMODE
* if proceeding visit: www.shop.com -> www.shop.com/store/STOREID/sm/SHOPPINGMODE

`/products`
* If first time visit: www.shop.com/products -> choose store -> www.shop.com/store/STOREID/sm/SHOPPINGMODE/products
* if proceeding visit: www.shop.com/products -> www.shop.com/store/STOREID/sm/SHOPPINGMODE/products

Hopefully that makes sense. What would be the most efficient way to implement this scenario? Middlewares? Redirects via config or in page? Something else?

Any help would be greatly appreciated!
